pkg/cache/v2/resource.go

```diff
@@ -54,6 +54,25 @@ func GetResponseType(typeURL string) types.ResponseType {
 	return types.UnknownType
 }
 
+// GetResponseTypeURL returns the type url for a valid enum
+func GetResponseTypeURL(responseType types.ResponseType) string {
+	switch responseType {
+	case types.Endpoint:
+		return resource.EndpointType
+	case types.Cluster:
+		return resource.ClusterType
+	case types.Route:
+		return resource.RouteType
+	case types.Listener:
+		return resource.ListenerType
+	case types.Secret:
+		return resource.SecretType
+	case types.Runtime:
+		return resource.RuntimeType
+	}
+	return resource.AnyType
```

Comment:
Ah yes, so I didn't know what else to default to since there isn't really a direct mapping to `types.UnknownType`. Would it make sense to just return an empty string here? Callers would just have to check to make sure the conversion was successful

Reply:
How about returning (string, error)? The behavior if there are multiple unknown types is unclear to me, maybe consider it an error for now?

Reply:
Now that I'm thinking, this function might just be invalid when we go to support custom types. I think like you say later on, just going off the known types in the snapshot is the way to go instead of trying to do conversion here.